download_from_csv.py

```diff
@@ -1,41 +1,150 @@
-'''Script to download your spotify playlists
+'''
+Script to download your spotify playlists:
 
-Convert your spotify playlists to csv from here: http://joellehman.com/playlist/
+Convert your spotify playlist to csv from here: http://joellehman.com/playlist
+or provide playlist URI, your username and authorization token here
 and then download all the songs through this script
 Depends upon youtube_dl, eyed3 and unidecode pip packages
 '''
 
+from __future__ import unicode_literals
 import os
 import csv
-import eyed3
 import argparse
+import json
+import UserString
+
+import eyed3
 import youtube_dl
 from unidecode import unidecode
-from __future__ import unicode_literals
+import requests
+from bs4 import BeautifulSoup
+
+import album_art
+
+# Youtube-dl options
+opts = {
+    'format': 'bestaudio/best',
+    'forcejson': True,
+    'postprocessors': [{
+        'key': 'FFmpegExtractAudio',
+        'preferredcodec': 'mp3',
+        'preferredquality': '256',
+    }],
+    'progress_hooks': '',
+    'logger': '',
+    'outtmpl': ''
+}
+
+
+class MyLogger(object):
+
+    def debug(self, msg):
+        pass
+
+    def warning(self, msg):
+        pass
 
+    def error(self, msg):
+        print(msg)
+
+
+def download_finish(d):
+    if d['status'] == 'finished':
+        print '\x1b[1A\x1b[2K'
+        print "\x1b[1A[\033[93mConverting\033[00m] %s" % d['filename']
+
+
+def generate_csv(URI, username, auth_token):
+    cmd = 'curl -X GET "https://api.spotify.com/v1/users/'\
```

Comment:
Why are you taking such long route i.e executing a command (curl) through python, writing the output to file and then parsing the file?
(This also assumes that the user will have cURL installed, which is an added dependency)

Why not use requests to send the request and then parse the response?